### PR TITLE
update user settings controllers to new permission system [fixes SCI-…

### PR DESCRIPTION
**Pull request by ZmagoD**
…1960]
 https://biosistemika.atlassian.net/browse/SCI-1960

### CODE REVIEW

File: app/controllers/users/invitations_controller.rb [hunk 1]
```diff
@@ -192,7 +192,7 @@ def check_invite_users_permission
       @role = params['role']
 
       render_403 if @emails && @emails.empty?
-      render_403 if @team && !is_admin_of_team(@team)
+      render_403 if @team && !can_read_team?(@team)
```
Comment (mlorb) [inline app/controllers/users/invitations_controller.rb]: Here goes `can_manage_team_users?`

File: app/controllers/users/settings/teams_controller.rb [hunk 2]
```diff
@@ -147,7 +147,7 @@ def load_user
 
       def load_team
         @team = Team.find_by_id(params[:id])
-        render_403 unless is_admin_of_team(@team)
+        render_403 unless can_read_team?(@team)
```
Comment (mlorb) [inline app/controllers/users/settings/teams_controller.rb]: I would remove this check from `load_team` and add separate method `check_update_team_permission`, where you would check `can_update_team?(@team)`

File: app/permissions/team.rb [hunk 3]
```diff
@@ -117,3 +117,19 @@
     can_create_repository_columns?(user, repository_column.repository.team)
   end
 end
+
```
Comment (mlorb) [inline app/permissions/team.rb]: Ne rabaš :)

File: app/controllers/users/settings/user_teams_controller.rb [hunk 4]
```diff
@@ -150,7 +150,7 @@ def load_user_team
         # Don't allow the user to modify UserTeam-s if he's not admin,
         # unless he/she is modifying his/her UserTeam
         if current_user != @user_t.user &&
-           !is_admin_of_team(@user_t.team)
+           !can_read_team?(@user_t.team)
```
Comment (mlorb) [inline app/controllers/users/settings/user_teams_controller.rb]: Here the same, I would make separate methods `check_leave_team_permission` and `check_manage_user_team_permission`.

File: app/controllers/users/settings/teams_controller.rb (hunk 2 above)
Comment (ZmagoD) [reply app/controllers/users/settings/teams_controller.rb]: but there is a banana. Until now we didn't show this controller actions if the user was not an admin

Comment (ZmagoD) [reply app/controllers/users/settings/teams_controller.rb]: Should I make a new permission check like can_manage_team?

Comment (mlorb) [reply app/controllers/users/settings/teams_controller.rb]: Just use then `can_update_team?(@team)`

File: app/controllers/users/settings/user_teams_controller.rb (hunk 4 above)
Comment (mlorb) [reply app/controllers/users/settings/user_teams_controller.rb]: And here use `can_manage_team_users?`

File: app/permissions/team.rb (hunk 3 above)
Comment (ZmagoD) [reply app/permissions/team.rb]: done

**Comment by mlorb** [comment]
Dropdown still needs to be fixed.

**Comment by ZmagoD** [comment]
This is a problem with canaid gem that evaluates can_create_teams to false. This issue will be probably fixed with this PR https://github.com/biosistemika/scinote-web/pull/968

**Comment by ZmagoD** [comment]
Try to remove all add-ons from your source tree

**Pull request merged**